### PR TITLE
chore: working on migrating instanceof examples to switch

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: examples/nodejs/cache/doc-example-files/config-and-error-handling.ts [hunk 1]
```diff
@@ -31,14 +39,17 @@ async function example_configuration_ConstructWithLambdaConfig() {
...
+      break;
+    case CacheGetResponse.Miss:
+      console.log("Key 'test-key' was not found in cache 'test-cache'");
+      break;
+    case CacheGetResponse.Error:
+      throw new Error(
+        `An error occurred while attempting to get key 'test-key' from cache 'test-cache': ${result.errorCode()}: ${result.toString()}`
+      );
   }
 }
```
Comment (cprice404) [inline examples/nodejs/cache/doc-example-files/config-and-error-handling.ts:55]: most of the changes are in the next file, which is hidden due to "large diffs" 🤦

File: examples/nodejs/cache/doc-example-files/config-and-error-handling.ts [hunk 2]
```diff
@@ -8,7 +8,15 @@
  * a second file in the future if desired.
  *
  */
-import {CacheClient, Configurations, CredentialProvider, CacheGet, SdkError, MomentoErrorCode} from '@gomomento/sdk';
+import {
+  CacheClient,
+  Configurations,
+  CredentialProvider,
+  CacheGet,
```
Comment (malandis) [inline examples/nodejs/cache/doc-example-files/config-and-error-handling.ts:15]:
```suggestion
```

GitHub lint complains this is never used.

File: examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts [hunk 3]
```diff
@@ -451,162 +481,192 @@ async function example_API_GetBatch(cacheClient: CacheClient, cacheName: string)
...
-  } else if (result instanceof CacheListFetch.Miss) {
-    console.log(`List 'test-list' was not found in cache '${cacheName}'`);
-  } else if (result instanceof CacheListFetch.Error) {
-    throw new Error(
-      `An error occurred while attempting to fetch the list 'test-list' from cache '${cacheName}': ${result.errorCode()}: ${result.toString()}`
-    );
+  switch (result.type) {
+    case CacheListFetchResponse.Hit:
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      console.log(`List fetched successfully: ${result.valueList()}`);
```
Comment (malandis) [inline examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:487]: Is there a way to write this line so we can remove the eslint suppression from the docs?

File: examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts [hunk 4]
```diff
@@ -964,34 +1074,40 @@ async function example_API_SortedSetRemoveElements(cacheClient: CacheClient, cac
...
   const result = await cacheClient.keyExists(cacheName, 'test-key');
-  if (result instanceof CacheKeyExists.Success) {
-    console.log("Does 'test-key' exists in the cache?", result.exists());
-  } else if (result instanceof CacheKeyExists.Error) {
-    throw new Error(
-      `An error occurred while attempting to call keyExists on key 'test-key' in cache '${cacheName}': ${result.errorCode()}: ${result.toString()}`
-    );
+  switch (result.type) {
+    case CacheKeyExistsResponse.Success:
+      console.log("Does 'test-key' exists in the cache?", result.exists());
```
Comment (malandis) [inline examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:1092]:
```suggestion
      console.log("Does 'test-key' exist in the cache?", result.exists());
```

File: examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts [hunk 5]
```diff
@@ -1234,22 +1389,24 @@ async function example_API_TopicSubscribe(topicClient: TopicClient, cacheName: s
...
+      await new Promise(resolve => setTimeout(resolve, 1000));
 
-    // Need to close the stream before the example ends or else the example will hang.
-    result.unsubscribe();
-  } else if (result instanceof TopicSubscribe.Error) {
-    throw new Error(
-      `An error occurred while attempting to subscribe to the topic 'test-topic' in cache '${cacheName}': ${result.errorCode()}: ${result.toString()}`
-    );
+      // Need to close the stream before the example ends or else the example will hang.
+      result.unsubscribe();
```
Comment (malandis) [inline examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:1404]: See my comment about the parentheticals to ourselves vs users

**Review by malandis — COMMENTED** [review]
Caught a couple small issues.

A non-blocker but for us to consider:
- Some of the inline comments are a guide to our users, eg in the topics examples where we set up a subscription, then publish.
- Others are comments that look more like to us. Can consider adding some metadata to the comment or making a comment "region" that is excluded from the docs

File: examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts (hunk 5 above)
Comment (cprice404) [reply examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:1404]: this seems useful for users too though?

**Review by malandis — APPROVED** [review]

**Pull request merged**